tests/test_api.py

```diff
@@ -465,11 +463,17 @@ def test_metadata_root(self):
         root.signed.add_key('root', key_metadata)
         self.assertEqual(pre_add_keyid, root.signed.roles['root'].keyids)
 
-        # Remove the key
-        root.signed.remove_key('root', keyid)
+        # Add the same key to targets as well
+        root.signed.add_key('targets', key_metadata)
 
-        # Assert that root does not contain the new key anymore
+        # Remove the key from root
+        root.signed.remove_key('root', keyid)
         self.assertNotIn(keyid, root.signed.roles['root'].keyids)
+        self.assertIn(keyid, root.signed.keys)
+
+        # Remove the key from targets as well
+        root.signed.remove_key('targets', keyid)
+        self.assertNotIn(keyid, root.signed.roles['targets'].keyids)
```

Comment:
And optionally one comment before `self.assertNotIn(...)` here as well:
`key is not used by any role and should have been deleted from keys`

Reply:
this one I did not do yet, I think the comment is pretty clear: key has now been removed from both roles. If the purpose of remove_key() is not clear then let's document that in the API reference not in the tests.

Reply:
Agreed that this one is better and it's okay without a change. That's why I said `optionally` probably should have said `consider` as our favorite word :D